src/LndClient/Data/HtlcEvent.hs

```diff
@@ -13,16 +14,32 @@ data HtlcEvent
         incomingHtlcId :: Word64,
         outgoingHtlcId :: Word64,
         timestampNs :: Word64,
-        eventType :: Enumerated GRPC.HtlcEvent_EventType
+        eventType :: EventType
       }
   deriving (Eq)
 
-instance FromGrpc HtlcEvent GRPC.HtlcEvent where
+data EventType
+  = UNKNOWN
+  | SEND
+  | RECEIVE
+  | FORWARD
+  deriving (Eq)
+
+instance FromGrpc HtlcEvent LnGRPC.HtlcEvent where
   fromGrpc x =
     HtlcEvent
-      <$> fromGrpc (GRPC.htlcEventIncomingChannelId x)
-      <*> fromGrpc (GRPC.htlcEventOutgoingChannelId x)
-      <*> fromGrpc (GRPC.htlcEventIncomingHtlcId x)
-      <*> fromGrpc (GRPC.htlcEventOutgoingHtlcId x)
-      <*> fromGrpc (GRPC.htlcEventTimestampNs x)
-      <*> fromGrpc (GRPC.htlcEventEventType x)
+      <$> fromGrpc (x ^. LnGRPC.incomingChannelId)
```

Comment:
Its actually could be changed to fold. So you don't right fromGrpc all the time

Reply:
decided to postpone putting out fromGrpc